packages/components/src/components/data-list/data-list.tsx

```diff
@@ -44,196 +44,183 @@ type DataListProps = {
-            </CellMeasurer>
-        ) : (
-            <div key={row_key} style={style}>
-                {getContent()}
-            </div>
-        );
-    };
-    let timeout: ReturnType<typeof setTimeout>;
-    const handleScroll = (e: ScrollParams): void => {
-        const { scrollTop } = e;
-
-        clearTimeout(timeout);
-        if (!is_scrolling) {
-            setIsScrolling(true);
-        }
-        timeout = setTimeout(() => {
-            if (!is_loading) {
-                setIsScrolling(false);
+        React.useEffect(() => {
+            if (is_dynamic_height) {
+                list_ref.current?.recomputeGridSize?.({ columnIndex: 0, rowIndex: 0 });
             }
-        }, 200);
+            trackItemsForTransition();
+        }, [data_source, is_dynamic_height, trackItemsForTransition]);
+
+        const footerRowRenderer = () => {
+            return <React.Fragment>{other_props.rowRenderer({ row: footer, is_footer: true })}</React.Fragment>;
+        };
+
+        const rowRenderer = ({ style, index, key, parent }: ListRowProps) => {
+            const { getRowAction, passthrough, row_gap } = other_props;
+            const row = data_source[index];
+            const action = getRowAction && getRowAction(row);
+            const destination_link = typeof action === 'string' ? action : undefined;
+            const action_desc = typeof action === 'object' ? action : undefined;
+            const row_key = keyMapper?.(row) || key;
+
+            const getContent = ({ measure }: GetContentType = {}) => (
+                <DataListRow
+                    action_desc={action_desc}
+                    destination_link={destination_link}
+                    is_new_row={!items_transition_map_ref.current[row_key]}
+                    is_scrolling={is_scrolling}
+                    measure={measure}
+                    passthrough={passthrough}
+                    row_gap={row_gap}
+                    row_key={row_key}
+                    row={row}
+                    rowRenderer={other_props.rowRenderer}
+                    is_dynamic_height={is_dynamic_height}
+                />
+            );
+
+            return is_dynamic_height && cache.current ? (
+                <CellMeasurer cache={cache.current} columnIndex={0} key={row_key} rowIndex={index} parent={parent}>
+                    {({ measure }) => <div style={style}>{getContent({ measure })}</div>}
+                </CellMeasurer>
+            ) : (
+                <div key={row_key} style={style}>
+                    {getContent()}
+                </div>
+            );
+        };
 
-        setScrollTop(scrollTop);
-        onScroll?.(e);
-    };
+        const handleScroll: React.UIEventHandler<HTMLDivElement> = ev => {
+            let timeout;
 
-    const setRef = (ref: MeasuredCellParent) => {
-        list_ref.current = ref;
-        setListRef?.(ref);
-    };
+            clearTimeout(timeout);
```

Comment:
why do we need this `clearTimeout(timeout)` immediately after the declaration of `timeout` variable?
@niloofar-deriv can you please take a look?

Reply:
this has been notified in the sonarcloud quality check

Reply:
Thanks for reviewing :pray: 
we need that for clearing the timeout :thinking: 
![Uploading image.png…]()

Reply:
As we talked we will fix it in a separate card :pray: